### PR TITLE
CORCI-1042 test: Reenable hdf5/vol unit tests

### PR DESCRIPTION
**Pull request by mjean308**
This PR will re-enable the hdf5 vol daos tests with HDF5 1.13.0-rc5 and HDF5 VOL DAOS 1.1.0rc1

The following bugs should be fixed by this update:
DAOS-6427 H5_DAOS_REMOTE_ERROR in IorSmall.
DAOS-5831 Daos Vol container test1 failed with error in hdf5_testpar/t_shapesame.c
DAOS-5469  HDF5 vol daos unit test h5_test_testhdf5 fails
DAOS-5647 HDF5 VOL h5vl_test_parallel failed with "data verification failed for chunk 9"
DAOS-6076 HDF5 vol daos unit test h5_partest_testhdf5 fails with mpich only
DAOS-5496 HDF5 vol daos unit test h5_partest_t_bigio fails

PR-repo: hdf5@PR-20 hdf5-vol-daos@PR-11 ior-hpc@PR-13 MACSio@PR-7
Skip-unit-tests: true
Skip-func-hw-test-medium: true
Test-tag-hw-large: macsio_daos_vol iorsmall soak_smoke macsio_daos vol hdf5

Signed-off-by: Maureen Jean <maureen.jean@intel.com>

### CODE REVIEW

**Comment by brianjmurrell** [comment]
Please retry this with `ior-hpc@PR-13 MACSio@PR-7` removed from `PR-repos:`.  Those should not need a rebuild.  Shared libraries take care of this for us.

And note that your commit message had a typo in that pragma.  It had: `PR-repo:`.

**Review by shimizukko — APPROVED** [review]
LGTM

**Review by saurabhtandan — DISMISSED** [review]
LGTM

File: src/tests/ftest/io/ior_small.py [hunk 1]
```diff
@@ -37,42 +37,40 @@ def test_ior_small(self):
...
+
         for oclass in obj_class:
             self.ior_cmd.dfs_oclass.update(oclass)
             for api in apis:
                 if api == "HDF5-VOL":
                     self.ior_cmd.api.update("HDF5")
                     hdf5_plugin_path = self.params.get(
                         "plugin_path", '/run/hdf5_vol/*')
+                    if "-k" not in flags:
+                        self.ior_cmd.flags.update(flags.append("-k"))
```
Comment (phender) [inline src/tests/ftest/io/ior_small.py:56]:
```suggestion
                        flags_w_k = " ".join([flags[0]] + ["-k"])
                        self.ior_cmd.flags.update(flags_w_k)
```

File: src/tests/ftest/io/ior_small.py [hunk 2]
```diff
@@ -37,42 +37,40 @@ def test_ior_small(self):
...
-                        self.log.info(
-                            "** SKIP test case: api=%s (transfer=%s, block=%s) "
-                            "for DAOS-6427", api, test[0], test[1])
-                        cncl_tickets.append("DAOS-6427")
-                        results.append(["CANCEL", str(self.ior_cmd)])
-                        continue
                     # run ior
+                    if api == "HDF5-VOL":
+                        sub_dir = get_random_string(5)
+                        dfuse_mount_dir = "/".join(["/tmp/daos_dfuse", sub_dir])
```
Comment (phender) [inline src/tests/ftest/io/ior_small.py:69]:
```suggestion
                        mount_dir = self.params.get("mount_dir", "/run/dfuse/*")
                        dfuse_mount_dir = os.path.join(mount_dir, sub_dir)
```

**Review by phender — CHANGES_REQUESTED** [review]

**Review by phender — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM

**Review by shimizukko — APPROVED** [review]
LGTM

**Comment by mjean308** [comment]
The single failure is due to a timeout in daos_core_test Epoch Recovery Test 15.   

This is a known failure - DAOS-6590.    However DAOS-6590  was duped to DAOS-6544 which only disabled daos_test -m, not test 15 so this test continues to run in CI.   It appears to be intermittent.

**Pull request merged**